syntax = "proto2";

import "common/complex_number.proto";

package magnon.common;

message MatrixXd {
    optional uint32 num_rows = 1;
    optional uint32 num_columns = 2;
    repeated double entry = 3;
}

message MatrixXi {
    optional uint32 num_rows = 1;
    optional uint32 num_columns = 2;
    repeated int32 entry = 3;
}

message MatrixXcd {
    optional uint32 num_rows = 1;
    optional uint32 num_columns = 2;

    repeated ComplexNumber entry = 3;
}

message Vector4d {
    optional double entry_0 = 1;
    optional double entry_1 = 2;
    optional double entry_2 = 3;
    optional double entry_3 = 4;
}

message Matrix4d {
    optional Vector4d row_0 = 1;
    optional Vector4d row_1 = 2;
    optional Vector4d row_2 = 3;
    optional Vector4d row_3 = 4;
};
